sequencer/datastream.go

```diff
@@ -0,0 +1,48 @@
+package sequencer
+
+import (
+	"time"
+	"unsafe"
+
+	"github.com/0xPolygonHermez/zkevm-data-streamer/datastreamer"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+const (
+	// StreamTypeSequencer represents a Sequencer stream
+	StreamTypeSequencer datastreamer.StreamType = 1
+	// EntryTypeL2Block represents a L2 block
+	EntryTypeL2Block datastreamer.EntryType = 1
+	// EntryTypeL2Tx represents a L2 transaction
+	EntryTypeL2Tx datastreamer.EntryType = 2
+)
+
+// L2Block represents a L2 block
+type L2Block struct {
+	BatchNumber    uint64
+	L2BlockNumber  uint64
+	Timestamp      time.Time
+	GlobalExitRoot common.Hash
+	Coinbase       common.Address
+}
+
+// Encode returns the encoded L2Block as a byte slice
+func (b L2Block) Encode() []byte {
+	const size = int(unsafe.Sizeof(L2Block{}))
+	return (*(*[size]byte)(unsafe.Pointer(&b)))[:]
```

Comment:
Are you sure about using this package right?? Isn't there other way to do it? It is a bit scary
Maybe this is helpful https://stackoverflow.com/questions/16330490/in-go-how-can-i-convert-a-struct-to-a-byte-array

Reply:
This is the easiest and faster way to convert a struct into a []byte. I know unsafe sounds "unsafe" but it is there for a reason and it only means YOU must take care of making it safe, in the same way unsafe works in Rust.

IMHO in this case where the only thing we use it for is to get a slice of consecutive memory into a slice. It is safe. There is no arithmetic involved or anything harmful.

Also there are parts of the code in the library that use reflection and it is though to be used also with real (unsafe) pointers.